packages/ibm-products/src/components/StringFormatter/StringFormatter.js

```diff
@@ -0,0 +1,176 @@
+ * overlay on hover or focus with the entirety of the provided copy.
+ */
+export let StringFormatter = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+
+      className,
+      lines = defaults.lines,
+      tooltipDirection = defaults.tooltipDirection,
+      truncate = defaults.truncate,
+      width = defaults.width,
+      value,
+      /* TODO: add other props for StringFormatter, with default values if needed */
+
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    return (
+      <span
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className // Apply any supplied class names to the main HTML element.
+        )}
+        ref={ref}
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        {...getDevtoolsProps(componentName)}
+      >
+        {truncate ? (
+          <DefinitionTooltip
+            className={`${blockClass}__tooltip`}
+            align={tooltipDirection}
+            definition={value}
+            openOnHover={true}
+          >
+            <StringFormatterContent
+              lines={lines}
+              truncate={truncate}
+              value={value}
+              width={width}
+            />
+          </DefinitionTooltip>
+        ) : (
+          <StringFormatterContent
+            className={className}
+            lines={lines}
+            truncate={truncate}
+            value={value}
+            width={width}
+          />
+        )}
+      </span>
+    );
+  }
+);
+
+// Return a placeholder if not released and not enabled by feature flag
+StringFormatter = pkg.checkComponentEnabled(StringFormatter, componentName);
+
+// The display name of the StringFormatter component, used by React. Note that displayName
+// is used in preference to relying on function.name.
+StringFormatter.displayName = componentName;
+
+const DefaultStringFormatterProps = {
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className: PropTypes.string,
+
+  /** Number of lines to clamp value. */
+  lines: PropTypes.number,
+  /** Whether or not the value should be truncated. */
+  truncate: PropTypes.bool,
+  /** Value to format. */
+  value: PropTypes.string.isRequired,
+  /** Maximum width of value which should include */
+  width: PropTypes.string,
+};
+
+// The types and DocGen commentary for the component props,
+// in alphabetical order (for consistency).
+// See https://www.npmjs.com/package/prop-types#usage.
+StringFormatter.propTypes = {
+  ...DefaultStringFormatterProps,
+
+  /** Specify the direction of the tooltip. Can be either top or bottom. */
+  tooltipDirection: PropTypes.oneOf(Object.values(StringFormatterAlignment)),
+};
+
+/**
+ * StringFormatterContent
+ * Only used internally
+ */
+
+let StringFormatterContent = React.forwardRef(
```

Comment:
as a general convention we should always separate components into their own files. this seems small enough that i don't think it's strictly _necessary_ to have it's own dedicated file. i think this code could just be lumped back into `StringFormatter`. either way is completely fine.